source/slang/parser.cpp

```diff
@@ -2210,8 +2217,8 @@ namespace Slang
         addModifier(bufferVarDecl, reflectionNameModifier);
 
         // Both the buffer variable and its type need to have names generated
-        bufferVarDecl->nameAndLoc.name = generateName(parser, "parameterGroup_" + reflectionNameToken.Content);
-        bufferDataTypeDecl->nameAndLoc.name = generateName(parser, "ParameterGroup_" + reflectionNameToken.Content);
+        bufferVarDecl->nameAndLoc.name = generateName(parser, "parameterGroup_" + String(reflectionNameToken.Content));
```

Comment:
We are *almost* at the point where we could start trying to pull out all this mess around synthesizing unique names for `cbuffer` and `tbuffer` declarations in user code.

The main reason those are needed is so that if you pass both `foo.vs.hlsl` and `foo.ps.hlsl` into Slang, and they both declare `cbuffer C { ... }` we can detect that those constant buffers are meant to represent the "same" declaration in both files (parameter binding/layout will only compute one layout for both declarations). In the end the original name is being preserved in a modifier, so we could just make the layout logic smart enough to check that instead.

This has nothing to do with this change, but I just wanted to note that the code here is due for a cleanup.